src/test/groovy/graphql/execution/reactive/ReactiveSupportTest.groovy

```diff
@@ -0,0 +1,222 @@
+    }
+
+    def "can get a reactive or flow publisher and make a CF from it"() {
+
+        when:
+        CompletableFuture<?> cf = ReactiveSupport.fetchedObject(reactiveObject) as CompletableFuture<?>
+
+        then:
+        !cf.isCancelled()
+        !cf.isCompletedExceptionally()
+        cf.isDone()
+
+        cf.join() == "X"
+
+        where:
+        reactiveObject         || _
+        Mono.just("X")         || _
+        toFlow(Mono.just("X")) || _
+    }
+
+    def "can get a reactive or flow publisher that takes some time and make a CF from it"() {
+
+        when:
+        CompletableFuture<?> cf = ReactiveSupport.fetchedObject(reactiveObject) as CompletableFuture<?>
+
+        then:
+        !cf.isCancelled()
+        !cf.isCompletedExceptionally()
+        !cf.isDone()
+
+        cf.join() == "X"
+
+        where:
+        reactiveObject                || _
+        delayedMono("X", 100)         || _
+        toFlow(delayedMono("X", 100)) || _
+    }
+
+    def "can get a reactive or flow publisher with an error and make a CF from it"() {
+
+        when:
+        CompletableFuture<?> cf = ReactiveSupport.fetchedObject(reactiveObject) as CompletableFuture<?>
+
+        then:
+        !cf.isCancelled()
+        cf.isCompletedExceptionally()
+        cf.isDone()
+
+        when:
+        cf.join()
+
+        then:
+        def e = thrown(CompletionException.class)
+        e.cause.message == "Bang!"
+
+        where:
+        reactiveObject                                    || _
+        Mono.error(new RuntimeException("Bang!"))         || _
+        toFlow(Mono.error(new RuntimeException("Bang!"))) || _
+    }
+
+    def "can get a empty reactive or flow publisher and make a CF from it"() {
+
+        when:
+        CompletableFuture<?> cf = ReactiveSupport.fetchedObject(reactiveObject) as CompletableFuture<?>
+
+        then:
+        !cf.isCancelled()
+        !cf.isCompletedExceptionally()
+        cf.isDone()
+
+        cf.join() == null
+
+        where:
+        reactiveObject         || _
+        Mono.empty()         || _
+        toFlow(Mono.empty()) || _
+    }
+
+
+    def "can get a reactive or flow publisher but cancel it before a value turns up"() {
+
+        when:
+        CompletableFuture<?> cf = ReactiveSupport.fetchedObject(reactiveObject) as CompletableFuture<?>
+
+        then:
+        !cf.isCancelled()
+        !cf.isCompletedExceptionally()
+        !cf.isDone()
+
+        when:
+        def cfCancelled = cf.cancel(true)
+
+        then:
+        cfCancelled
+        cf.isCancelled()
+        cf.isCompletedExceptionally()
+        cf.isDone()
+
+        when:
+        cf.join()
+
+        then:
+        thrown(CancellationException.class)
+
+        where:
+        reactiveObject                 || _
+        delayedMono("X", 1000)         || _
+        toFlow(delayedMono("X", 1000)) || _
+
+    }
+
+
+    def "can get a reactive Flux and only take one value and make a CF from it"() {
```

Comment:
even a Flux of N values only has 1 value taken from it because that is what the graphql contact is - a Query field is 1 value not  a stream of values - at least not yet (aka @stream future work perhaps)